lisa/_assets/kmodules/lisa/fs.c

```diff
@@ -0,0 +1,755 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#include <linux/fs.h>
+#include <linux/fs_context.h>
+#include <linux/seq_file.h>
+#include <linux/slab.h>
+#include <linux/user_namespace.h>
+#include <linux/percpu-rwsem.h>
+
+#include "features.h"
+#include "configs.h"
+
+static int lisa_fs_create_files(struct dentry *dentry, bool is_top_level, struct lisa_cfg *cfg);
+static struct dentry *
+lisa_fs_create_single(struct dentry *parent, const char *name,
+		      const struct inode_operations *i_ops,
+		      const struct file_operations *f_ops, umode_t mode,
+		      void *data);
+
+#define LISA_FS_SUPER_MAGIC 0xcdb11bc9
+
+/* Protect the interface. */
+static struct mutex interface_lock;
```

Comment:
Any reason for not using fs specific data (s_fs_info) for that ?
Is that mutex there to serialize access to fs nodes or to protect the config list ? or both

Reply:
I think it was necessary when the mutex was taken in `lisa_fs_fill_super()`.

It's not the case anymore, so I'm moving the mutex in `s_fs_info`, along with the global `cfg_list`